src/spin_mutex.cc

```diff
@@ -3,19 +3,40 @@
 #include <cstddef>
 #include <thread>
 
+#define _ASM_NOOP asm volatile("nop")
+
+// Cannot use `do { } while (0)` here because there would be huge differences between -O0 and -O3
```

Comment:
It's probably already huge between O0 and O1

Reply:
Yes